setup.py

```diff
@@ -25,6 +25,7 @@
     author_email='russell@keith-magee.com',
     url='http://pybee.org/rubicon',
     packages=find_packages(exclude=['tests']),
+    python_requires='>=3.5',
```

Comment:
I don't have any strong opinions on minimum Python versions, but is there any particular reason why we should require 3.5? AFAIK, the current minimum Python version for Rubicon is 3.4, and I'd be inclined to leave it at that unless there's a reason to drop 3.4. CPython 3.4 seems to still be supported, 3.4.8 was released a couple of months ago, on 2018-02-05.

If we decide to bump the minimum Python version, I think it would be better to go straight to 3.6, which has some very useful features like f-strings and ordered kwargs. It's probably too early to require the absolute latest version of Python though.

Adding `python_requires` is a good idea though, regardless of what minimum version we decide on, so users get an error when they try to `pip install` on an unsupported Python version.

Reply:
I don't think there's any specific motivation, other than the fact that 3.4 is now reaching end of life, and supporting less things is generally easier. 3.4 is in security-only mode now; when 3.4.8 was released, only sources were published. 

Rubicon is in a good position in that it doesn't need to use the newer language features, but is mature enough that porting across versions isn't that big a deal. Some of the other BeeWare projects (Toga, VOC and Batavia most notably) are a lot more sensitive to version changes, and dropping 3.4 specifically opens up some options (specifically, being able to use the `async`/`await` keywords). 

So - in the case of Rubicon, I'd tend to agree that it's worth keeping 3.4 support, at least until it is either (a) fully EOL'd, or (b) becomes a maintenance hurdle for some other reason.